### ISSUE TITLE
[Remote-SSH Bug]: Client refused: version mismatch

### ISSUE THREAD

**Issue by pfuntner** [microsoft/vscode-remote-release#9047]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [X] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It connects successfully

### Expected Behavior

I expect that vscode will show me a directory of my remote system.

### Steps To Reproduce

1. My laptop is Windows 11
2. My remote system is Ubuntu 20 openstack VM
3. Windows has OpenSSH, I've configured `.ssh/config`, and can use `ssh ubuntu` from a Windows command prompt and reach the remote system
4. Windows has vscode:
    ```
    Version: 1.82.2 (system setup)
    Commit: abd2f3db4bdb28f9e95536dfa84d8479f1eb312d
    Date: 2023-09-14T05:55:25.390Z
    Electron: 25.8.1
    ElectronBuildId: 23779380
    Chromium: 114.0.5735.289
    Node.js: 18.15.0
    V8: 11.4.183.29-electron.0
    OS: Windows_NT x64 10.0.22621
    ```
5. Windows has Remote - SSH v0.106.4
6. There is a firewall on Ubuntu machine that prevents it from downloading the the remote code.  I followed steps at [_How can I install vs-code-server manually and tell vs-code-remote?_](https://stackoverflow.com/questions/56073276/how-can-i-install-vs-code-server-manually-and-tell-vs-code-remote) to install `vscode-server-linux-x64.tar.gz` for `abd2f3db4bdb28f9e95536dfa84d8479f1eb312d`:
    ```
    $ find ~/.vscode-server -maxdepth 2 | xargs ls -1dF | indent 4
    /home/ubuntu/.vscode-server/
    /home/ubuntu/.vscode-server/.abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.log
    /home/ubuntu/.vscode-server/.abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.pid
    /home/ubuntu/.vscode-server/.abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.token*
    /home/ubuntu/.vscode-server/bin/
    /home/ubuntu/.vscode-server/bin/abd2f3db4bdb28f9e95536dfa84d8479f1eb312d/
    /home/ubuntu/.vscode-server/data/
    /home/ubuntu/.vscode-server/data/Machine/
    /home/ubuntu/.vscode-server/data/User/
    /home/ubuntu/.vscode-server/data/logs/
    /home/ubuntu/.vscode-server/data/machineid
    /home/ubuntu/.vscode-server/extensions/
    /home/ubuntu/.vscode-server/extensions/extensions.json
    $
    ```
    log file: [.abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.log](https://github.com/microsoft/vscode-remote-release/files/12767531/default.abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.log)

7. When I try to open a remote directory from vscode, I get:
<img width="956" alt="image" src="https://github.com/microsoft/vscode-remote-release/assets/23261559/0f485171-e6df-4085-8539-fe46d973bb43">

### Remote-SSH Log

<details>
<summary>Remote-SSH Log</summary>
<p>

```

                                     WARNING!!!
                        READ THIS BEFORE ATTEMPTING TO LOGON

          This System is for the use of authorized users only.  Individuals
          using this computer without authority, or in excess of their
          authority, are subject to having all of their activities on this
          system monitored and recorded by system personnel.  In the course
          of monitoring individuals improperly using this system, or in the
          course of system maintenance, the activities of authorized users
          may also be monitored.  Anyone using this system expressly
          consents to such monitoring and is advised that if such
          monitoring reveals possible criminal activity, system personnel
          may provide the evidence of such monitoring to law enforcement
          officials.

          Cisco Acceptable Use Policy:
          http://wwwin.cisco.com/c/cec/organizations/security-trust/infosec/policies.html

    800ef10854c5: running
    Acquiring lock on /home/ubuntu/.vscode-server/bin/abd2f3db4bdb28f9e95536dfa84d8479f1eb312d/vscode-remote-lock.ubuntu.abd2f3db4bdb28f9e95536dfa84d8479f1eb312d
    Found existing installation at /home/ubuntu/.vscode-server/bin/abd2f3db4bdb28f9e95536dfa84d8479f1eb312d...
    Checking /home/ubuntu/.vscode-server/.abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.log and /home/ubuntu/.vscode-server/.abd2f3db4bdb28f9e95536dfa84d8479f1eb312d.pid for a running server
    Looking for server with pid: 2676
    Found running server...

    *
    * Visual Studio Code Server
    *
    * By using the software, you agree to
    * the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-license) and
    * the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacystatement).
    *

    Checking server status on port 45193 with wget
    800ef10854c5: start
    SSH_AUTH_SOCK====
    DISPLAY====
    webUiAccessToken====
    listeningOn==45193==
    osReleaseId==ubuntu==
    arch==x86_64==
    vscodeArch==x64==
    bitness==64==
    tmpDir==/run/user/1000==
    platform==linux==
    unpackResult====
    didLocalDownload==0==
    downloadTime====
    installTime====
    extInstallTime====
    serverStartTime====
    connectionToken==a75ec3cd-ae41-4cdf-bf9e-4d7299f3091a==
    800ef10854c5: end
```

</p>
</details>

### Anything else?

I had things set up fine for a while.  I think there was an upgrade to vscode and/or the extension and I could no longer get to my remote system.  It was then that I noticed there was a firewall preventing download on the remote system but I still couldn't get it working after installing it manually.

I gave up for a while and, for unrelated reasons, I got a new Windows laptop, reinstalled vscode and the extension and now I'm trying to get vscode working again.  I think I'm pretty close and am encouraged that the banner on the remote system is present in the debugging output.

**Comment by roblourens** [comment]
Since you are installing the server manually, you will have to update it manually for every vscode release. The error is saying that the client version is out of sync with the server version. We don't really officially support this scenario, and it's a bit of a pain to maintain, but it should work for you as long as the server is installed properly.

**Comment by pfuntner** [comment]
> Since you are installing the server manually, you will have to update it manually for every vscode release. 

I understand.  I don't have much choice if I want to use vscode in my work environment.  Firewalls are not unusual and I wish this was supported.

> The error is saying that the client version is out of sync with the server version. We don't really officially support this scenario, and it's a bit of a pain to maintain, but it should work for you as long as the server is installed properly.

How is it improperly installed?  Can you advise?

**Comment by roblourens** [comment]
The error message says it's the wrong version. That's all I can tell from here.

**Comment by pfuntner** [comment]
I found a workaround!  I was lamenting the loss of VScode with my colleagues and sought out alternative IDEs.  They asked me what the issue was and I explained that I thought it was firewall preventing me from accessing some pieces of the outside world.  We face that problem for other tasks and are able to use a proxy to get past the firewall.  It occurred to me that maybe VScode would let me provide a proxy and use it to download the tarball on my remote target.

I found the proxy option, VScode installed on the target without a hiccup, and I'm back in bidness! 🤓
